jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -694,6 +693,58 @@ public Response getInvoicesForAccount(@PathParam("accountId") final UUID account
         return Response.status(Status.OK).entity(result).build();
     }
 
+    @TimedResource
+    @GET
+    @Path("/{accountId:" + UUID_PATTERN + "}/" + INVOICES + "/" + PAGINATION)
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Retrieve account invoices", response = InvoiceJson.class, responseContainer = "List")
+    @ApiResponses(value = {@ApiResponse(code = 400, message = "Invalid account id supplied"),
+                           @ApiResponse(code = 404, message = "Account not found")})
+    public Response getInvoicesForAccountWithPagination(@PathParam("accountId") final UUID accountId,
+                                          @QueryParam(QUERY_INVOICES_FILTER) final String invoicesFilter,
+                                          @QueryParam(QUERY_SEARCH_OFFSET) @DefaultValue("0") final Long offset,
+                                          @QueryParam(QUERY_SEARCH_LIMIT) @DefaultValue("100") final Long limit,
+                                          @QueryParam(QUERY_AUDIT) @DefaultValue("NONE") final AuditMode auditMode,
+                                          @javax.ws.rs.core.Context final HttpServletRequest request) throws AccountApiException {
+
+        final TenantContext tenantContext = context.createTenantContextWithAccountId(accountId, request);
+
+        // Verify the account exists
+        accountUserApi.getAccountById(accountId, tenantContext);
+
+        final Pagination<Invoice> invoices = invoiceApi.getInvoicesByAccount(accountId, offset, limit, tenantContext);
+
+        //TODO_1272: Delete this filter?
```

Comment:
Yes, I would remove it as the offset / count would be wrong in the Pagination object. If / When we want to introduce it, we should implement a `search` API instead (I don't suggest we introduce it just yet though).

Reply:
Fixed.